routes/author/index.js

```diff
@@ -19,59 +18,64 @@ module.exports = function(req, res){
       } finally {
         book.images = images;
       }
-      if(err){
+      if (err) {
         console.log(err);
         return book;
       }
       return book;
     });
-  };
+  }
 
-  var authorId = Number(req.params[0]);
-  var authors = {};
+  const authorId = Number(req.params[0]);
+  const authors = {};
+  if (isNaN(authorId)) {
+    return res.status(404).render('error', {
+      title: 'ページが見つかりません。',
+    });
+  }
 
-  var countAuthors = function(){
-    var d = Q.defer();
-    Author.count({}, function(err, count){
+  function countAuthors() {
+    const deffered = Q.defer();
+    Author.count({}, (err, count)=> {
       authors.count = count;
-      d.resolve(authors);
+      deffered.resolve(authors);
     });
-    return d.promise;
-  };
+    return deffered.promise;
+  }
 
-  var getAuthor = function(authorId, index){
-    var d = Q.defer();
-    Author.findOne({ pageId: authorId }, function(err, author){
-      if(err){
-        return console.log("err", err);
+  function getAuthor(id, index) {
+    const deffered = Q.defer();
+    Author.findOne({ pageId: id }, (err, author)=> {
+      if (err) {
+        return deffered.reject(err);
       }
       authors[index] = author;
-      d.resolve(authors);
+      deffered.resolve(authors);
     });
-    return d.promise;
-  };
+    return deffered.promise;
+  }
 
-  var getAuthorParallel = function(){
-    var authorIds = [
+  function getAuthorParallel() {
+    const authorIds = [
       authorId - 1,
       authorId,
-      authorId + 1
+      authorId + 1,
     ];
 
-    if(authorIds[2] > authors.count){
+    if (authorIds[2] > authors.count) {
       authorIds[2] = 1;
     }
-    if(authorIds[0] === 0){
+    if (authorIds[0] === 0) {
       authorIds[0] = authors.count;
     }
 
-    var d = Q.defer();
+    const deffered = Q.defer();
     Q.all(authorIds.map(getAuthor))
-    .done(function(authors){
-      d.resolve(authors);
+    .done((resultAuthors)=> {
```

Comment:
__[jshint]__
*  'arrow function syntax (=>)' is only available in ES6 (use esnext option).